integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,147 @@
+from loguru import logger
+from typing import Any, AsyncGenerator
+
+from port_ocean.utils import http_async_client
+
+
+WEBHOOK_NAME = "Port-Ocean-Events-Webhook"
+WEBHOOK_EVENTS = [
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+]
+
+
+class ClickupClient:
+    def __init__(self, clickup_personal_token: str) -> None:
+        self.clickup_url = "https://api.clickup.com/api/v2"
+        self.api_auth_header = {"Authorization": clickup_personal_token}
+        self.client = http_async_client
+        self.client.headers.update(self.api_auth_header)
+
+    async def create_webhook_events(self, app_host: str) -> None:
+        for team in await self.get_teams():
+            await self._create_team_webhook_events(app_host, team["id"])
+
+    async def _create_team_webhook_events(self, app_host: str, team_id: int) -> None:
+        webhook_target_app_host = f"{app_host}/integration/webhook"
+        webhooks_response = await self.client.get(
+            f"{self.clickup_url}/team/{team_id}/webhook"
+        )
+        webhooks_response.raise_for_status()
+        webhooks = webhooks_response.json()
+
+        existing_webhook = next(
+            (
+                webhook
+                for webhook in webhooks["webhooks"]
+                if webhook["endpoint"] == webhook_target_app_host
+            ),
+            None,
+        )
+
```

Comment:
this whole logic, should be outside of the client, and the client should only expost `get` and `create` methods

Reply:
the resolution logic shouldn't be inside the client